workshop installer to support tls skip flag

ganchandrasekaran opened the pull request:
Installer to support TLS verification skip flag

ganchandrasekaran committed:
Update README.md

ganchandrasekaran committed:
clab install insecure

ganchandrasekaran committed:
Download cni archive with tls skip option

ganchandrasekaran committed:
download kind binary with tls skip suport

ganchandrasekaran committed:
download kpt binary with tls skip support

ganchandrasekaran committed:
create repo with tls skip support

shailendrarjpt commented on nephio-ansible-install/README.md:
```diff
@@ -38,11 +38,13 @@ all:
     github_username: <GitHub username>
     github_token: <GitHub personal access token>
-    github_organization: <optional, if a GitHub organization is used this should be filled out with the GitHub organization>
+    github_uri: github.com <replace this if you have a different github domain name in case of organization hosted github>
```
This change is to allow other github domain as well, mainly in case of any organization's internal git hub.
We need to make similar change on the UI as well to allow different git hub domains.

shailendrarjpt replied:
This property is added to allow other git hub domains as well, mainly in the case of organization specific git hub domains.

shailendrarjpt commented on nephio-ansible-install/README.md:
```diff
@@ -38,11 +38,13 @@ all:
     gitea_password: <gitea password>
     dockerhub_username: <dockerhub username>
     dockerhub_token: <dockerhub token or password>
+    validate_certs: true <change this to false if you want to avoid ssl/tls check>
```
This is been used in ansible's many pre requisite tasks to avoid ssl/tls checks if set to false.

shailendrarjpt commented on nephio-ansible-install/README.md:
```diff
@@ -99,9 +101,10 @@ ansible-galaxy collection install community.docker # required for gitea
 Now that the environment is up an running we can install the Nephio environment
 
 First we create some prerequisites, which installs kubectl, kind, kpt, cni and setup the bash environment
+-kK is for password promt to allow script to install dependencies.
```
This change is to allow user to enter the sudo password while running the ansible playbook else gets an error "sudo password is missing"

shailendrarjpt commented on nephio-ansible-install/README.md:
````diff
@@ -99,9 +101,10 @@ ansible-galaxy collection install community.docker # required for gitea
 
 ```bash
-ansible-playbook playbooks/install-prereq.yaml
+ansible-playbook playbooks/install-prereq.yaml -kK
````
This change is to allow user to enter the sudo password while running the ansible playbook else gets an error "sudo password is missing"

ganchandrasekaran replied:
yeah, this is needed when we execute from other user accounts

shailendrarjpt commented on nephio-ansible-install/roles/clab/tasks/main.yaml:
```diff
@@ -6,3 +6,60 @@
+  when: validate_certs
+
+
+- name: Create clab installation directory
```
This change is to add skip ssl/tls checks for all the curl commands and wget commands in the container lab installation script if the validate_certs is set to false.

henderiw commented:
lgtm

vjayaramrh commented on nephio-ansible-install/README.md:
```diff
@@ -38,11 +38,13 @@ all:
     github_token: <GitHub personal access token>
-    github_organization: <optional, if a GitHub organization is used this should be filled out with the GitHub organization>
+    github_uri: github.com <replace this if you have a different github domain name in case of organization hosted github>
+    github_organization: <if a GitHub organization is used this should be filled out with the GitHub organization, else provide your github username>
```
Which file is consuming the `githhub_organization`?

ganchandrasekaran replied:
The previous role `github_organisation` is not changed. As we only have introduced `github_uri` so to include other github domains as well, mainly in the case of organization specific git hub domains:
e.g. https://github.sec.xyzCompany.net/

ganchandrasekaran replied:
Thinking of this further, supporting different github domain name is out of scope for this PR. 
Will take it seperately in full. I have reverted related changes.

vjayaramrh commented:
@ganchandrasekaran A PR was recently merged, so you may want to resolve the merge conflicts as a first step

ganchandrasekaran committed:
Merge branch 'main' into ganchandrasekaran-patch-1

vjayaramrh commented on nephio-ansible-install/roles/clab/tasks/main.yaml:
```diff
@@ -6,3 +6,60 @@
+  when: validate_certs
+
+
+- name: Create clab installation directory
```
L12-L65 can be put in a single block.
refer https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_blocks.html

ganchandrasekaran replied:
Good point, Done

ganchandrasekaran committed:
Update README.md

Thinking about including different github domain name is out of scope for this commit. So it is better to remove from this PR and will create a seperate one in full.

ganchandrasekaran committed:
Update main.yaml

Supporting different github domain name is out of scope for this PR. Will take it seperately in full.

ganchandrasekaran committed:
Update main.yaml

included block based grouping

vjayaramrh commented:
/lgtm

vjayaramrh commented:
@henderiw @johnbelamaric @s3wong FYI
